src/components/ResoruceIcon.tsx

```diff
@@ -0,0 +1,47 @@
+import { Icon } from "@canonical/react-components";
+import { FC } from "react";
+
+export type ResourceIconType =
+  | "container"
+  | "virtual-machine"
+  | "snapshot"
+  | "profile"
+  | "project"
+  | "cluster-member"
+  | "network"
+  | "pool"
+  | "volume"
+  | "iso-volume"
+  | "image"
+  | "oidc-identity"
+  | "certificate"
+  | "auth-group"
+  | "device";
+
+export const resourceIcons: Record<ResourceIconType, string> = {
```

Comment:
```suggestion
const resourceIcons: Record<ResourceIconType, string> = {
```
I think we should keep the map a private concept and not export it.